src/firefly/js/ui/DownloadDialog.jsx

```diff
@@ -244,24 +261,24 @@ function hasProprietaryData(tableModel={}) {
     return false;
 }
 
-
 /**
- * creates and show the DownloadDialog.
- * @param {Component}  panel  the panel to show in the popup.
- * @param {boolean} [show=true] show or hide this dialog
+ *
+ * @param panel
+ * @param title
+ * @param show
  */
-function showDownloadDialog(panel, show=true) {
-    const ttl = panel.props.title || DOWNLOAD_DIALOG_ID;
+export function showDownloadDialog(panel, title, show=true) {
+
```

Comment:
Why did you remove logic here outside?  It's being called by several other places.  Were they all changed?

Reply:
Yes, I changed it in all places.  The DownloadButton was a class before.  Then this worked.  I don't know when and who changed to a stateless function.  After that change, the "this" keyword no longer worked.  Thus, the Dialog was not able to be closed as it reported in IRSA-3113.  If you know a better way to handle this, please feel free to push your changes.  THANKS!